operator/pkg/controllers/master/authenticatorconfig.go

```diff
@@ -35,146 +28,10 @@ func (c *Controller) reconcileAuthenticatorConfig(ctx context.Context, controlPl
 	if err != nil {
 		return fmt.Errorf("getting AWS account ID, %w", err)
 	}
-	configMap, err := object.GenerateConfigMap(authenticatorConfig, struct{ Name, ClusterName, Namespace, Group, KitNodeRole, AWSAccountID, PrivateDNS, SessionName string }{
-		Name:         AuthenticatorConfigMapName(controlPlane.ClusterName()),
-		ClusterName:  controlPlane.ClusterName(),
-		Namespace:    controlPlane.Namespace,
-		Group:        v1alpha1.SchemeGroupVersion.Group,
-		KitNodeRole:  iam.KitNodeRoleNameFor(controlPlane.ClusterName()),
-		AWSAccountID: awsAccountID,
-		PrivateDNS:   "{{EC2PrivateDNSName}}",
-		SessionName:  "{{SessionNameRaw}}",
-	})
-	if err != nil {
-		return fmt.Errorf("generating authenticator config, %w", err)
-	}
-	return c.kubeClient.EnsurePatch(ctx, &v1.ConfigMap{}, configMap)
+	return c.awsIamAuthenticator.EnsureConfig(ctx, controlPlane.ClusterName(), controlPlane.Namespace,
+		iam.KitNodeRoleNameFor(controlPlane.ClusterName()), awsAccountID)
```

Comment:
Potentially noop for this PR, but we should avoid using terms like KIT inside of KIT, as it makes renaming the project very had. I think iam.NodeRoleName(...) is plenty expressive.